src/Components/Diagnosis/ConsultationDiagnosisBuilder/AddICD11Diagnosis.tsx

```diff
@@ -36,6 +38,10 @@ export default function AddICD11Diagnosis(props: AddICD11DiagnosisProps) {
     }
   }, [res?.status]);
 
+  useEffect(() => props.prefill && setSelected(props.prefill), [props.prefill]);
+
+  useEffect(() => props.setPrefill?.(undefined), [selected]);
```

Comment:
@shivankacker can we remove this use effect and use a util function like handleSelection which updates both selected and prefill state